Month3/Project1/Project1/TDinamicArray.h

```diff
@@ -0,0 +1,290 @@
+#pragma once
+#include "Square.h"
+#include <stdlib.h>     
+#include <time.h> 
+
+
+template <typename T>
+class TDinamicArray
+{
+private:
+	int Capacity = 3;
+	int Size = 0;
+	T* Array = new T[Capacity];
+public:
+	TDinamicArray() 
+	{
+	
+	}
+	TDinamicArray(const int EntryCapacity)
+	{
+		Array = new T[EntryCapacity];
+	
+	}
+	~TDinamicArray() 
+	{
+		delete[] Array;
+	}
+
+	TDinamicArray(const TDinamicArray<T>& Array1)
+	{
+		Append(Array1);
+	}
+	
+
+	T& operator[](int Index)
+	{
+		return Array[Index];
+	}
+	const T& operator[](int Index)const
+	{
+		return Array[Index];
+	}
+
+
+
+	T& Front()
+	{
+		return Array[0];
+	}
+	const T& Front()const 
+	{
+		return Array[0];
+	}
+
+
+	T& Back()
+	{	
+		return Array[Size];
+	}
+	const T& Back() const
+	{
+		return Array[Size];
+	}
+
+
+	T& GetData()
+	{
+		return Array;
+	}
+	const T& GetData() const
+	{
+		return Array;
+	}
+
+	bool Empty()
+	{
+		return Capacity == 0;
+	}
+
+	int GetSize()
+	{
+		return Size;
+	}
+
+	int GetCapacity()
+	{
+		return Capacity;
+	}
+
+
+	void Reserve(int RequieredCapacity)
+	{
+		if (RequieredCapacity > Capacity)
+		{
+			T* NewArray = new T[RequieredCapacity];
+			for (int CurrentIndex = 0; CurrentIndex < Capacity; CurrentIndex++)
+			{
+				NewArray[CurrentIndex] = Array[CurrentIndex];
+			}
+			delete Array;
+
+			Capacity = RequieredCapacity;
+			Array = NewArray;
+		}
+
+		
+	}
+
+
+	void ShriktoFit()
+	{
+
+		T* NewArray = new T[Size];
+
+		for (int CurrentIndex = 0; CurrentIndex <= Size; CurrentIndex++)
+		{
+			NewArray[CurrentIndex] = Array[CurrentIndex];
+		}
+		delete Array;
+
+		Capacity = Size;
+		Array = NewArray;
+	}
+
+	void Clear()
+	{
+		if (Size > 0)
```

Comment:
I think this if clause isn't necessary. If it was already 0, removing this if clause would still not leave any changes, as it would set Size to be 0 again.